config/environments/development.rb

```diff
@@ -1,72 +1,81 @@
-  config.action_mailer.delivery_method = mailconf[:delivery_method]
-  config.action_mailer.smtp_settings = mailconf[:settings].clone
-  config.action_mailer.default_url_options = { :protocol => 'http://', :host => mailconf[:settings][:domain], :port => ':3000' }
-
-  MAIL_RECIPIENT_OVERRIDE = (ENV['OVERRIDE_RECIPIENTS']||[]).split(",")
-  
-  # Do not eager load code on boot.
+  # The test environment is used exclusively to run your application's
+  # test suite. You never need to work with it otherwise. Remember that
+  # your test database is "scratch space" for the test suite and is wiped
+  # and recreated between test runs. Don't rely on the data there!
+  config.cache_classes = true
+
+  # Do not eager load code on boot. This avoids loading your whole application
+  # just for the purpose of running a single test. If you are using a tool that
+  # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
-  # Show full error reports.
-  config.consider_all_requests_local = true
+  # Configure public file server with Cache-Control for performance.
+  config.public_file_server.enabled = true
+  config.public_file_server.headers = {
+    'Cache-Control' => "public, max-age=#{1.hour.to_i}"
+  }
 
-  # Enable/disable caching. By default caching is disabled.
-  # Run rails dev:cache to toggle caching.
-  if Rails.root.join('tmp', 'caching-dev.txt').exist?
-    config.action_controller.perform_caching = true
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  #config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
-    config.cache_store = :memory_store
-    config.public_file_server.headers = {
-      'Cache-Control' => "public, max-age=#{2.days.to_i}"
-    }
-  else
-    config.action_controller.perform_caching = false
+  # Show full error reports and disable caching.
+  config.consider_all_requests_local       = true
+  config.action_controller.perform_caching = false
 
-    config.cache_store = :null_store
-  end
+  # Raise exceptions instead of rendering exception templates.
+  config.action_dispatch.show_exceptions = false
+
+  # Disable request forgery protection in test environment.
+  #config.action_controller.allow_forgery_protection = false
 
-  # Store uploaded files on the local file system (see config/storage.yml for options)
   config.active_storage.service = :local
 
-  # Don't care if the mailer can't send.
-  config.action_mailer.raise_delivery_errors = false
+  # Use the lowest log level to ensure availability of diagnostic information when problems arise.
+  config.log_level = :debug
+
+  config.action_mailer.perform_caching = false
+
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :test delivery method accumulates sent emails in the
+  # ActionMailer::Base.deliveries array.
+  config.action_mailer.delivery_method = :smtp
+
+  config.action_mailer.smtp_settings = {
```

Comment:
This explicitly does not block approval if it's not gotten to, but:

The one concern I have with doing this rather than using the mailer.yml is that people outside LTS, or even our own future selves, might need a mailer with username/password/other mailer vars, and I think that having propertized config committed to the repo over a set of keys kind of... suggests that those are the only supported keys.  I might be off there, but I'd like maybe a comment here or in the README or both caveating this, and pointing to the mailer docs in railsguides?

Reply:
Good idea, I've added all of the available SMTP config values to the env-example.